### PR TITLE
Add force_fifo_present_mode optional setting

### PR DESCRIPTION
**Pull request by zongdu-arm**
If presentMode is forced to VK_PRESENT_MODE_FIFO_KHR in vkCreateSwapchainKHR, the application will crash if it does not use VK_PRESENT_MODE_FIFO_KHR presentMode when capturing.

### CODE REVIEW

**Comment by fabian-lunarg** [comment]
Ubuntu GCC Release failed due to clang-format issues in `framework/encode/capture_manager.h`  @zongdu-arm

**Comment by zongdu-arm** [comment]
Hi @fabian-lunarg ,
I have fixed the code check error, please approval to CI test again.

**Comment by fabian-lunarg** [comment]
> Hi @fabian-lunarg , I have fixed the code check error, please approval to CI test again.

"Author zongdu-arm not on autobuild list"
let me find out what we can do to un-block CI for your PRs, will come back

File: USAGE_android.md [hunk 1]
```diff
@@ -327,6 +327,7 @@ option values.
...
+| Force FIFO present mode                        | debug.gfxrecon.force_fifo_present_mode                        | BOOL    | When the `force_fifo_present_mode` is enabled, force all present modes in vkGetPhysicalDeviceSurfacePresentModesKHR to VK_PRESENT_MODE_FIFO_KHR, and app can be setted present mode in vkCreateSwapchain to VK_PRESENT_MODE_FIFO_KHR. Otherwise the original present mode will be used. Default is: `true`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 |
```
Comment (MarkY-LunarG) [inline USAGE_android.md:330]: NIt: "setted"?  maybe "app present mode is set in vkCreateSwapchain to VK_PRESENT_MODE_FIFO_KHR"?

**Review by MarkY-LunarG — APPROVED** [review]
Just one small nit to fix from my standpoint.

File: framework/encode/vulkan_capture_manager.cpp [hunk 2]
```diff
@@ -2711,9 +2711,17 @@ void VulkanCaptureManager::OverrideGetPhysicalDeviceSurfacePresentModesKHR(uint3
 {
     assert((pPresentModeCount != nullptr) && (pPresentModes != nullptr));
 
-    for (uint32_t i = 0; i < (*pPresentModeCount); ++i)
+    if (common_manager_->GetForceFifoPresentModeSetting())
     {
-        pPresentModes[i] = VK_PRESENT_MODE_FIFO_KHR;
+        GFXRECON_LOG_WARNING_ONCE(
+            "Force all present modes in vkGetPhysicalDeviceSurfacePresentModesKHR to VK_PRESENT_MODE_FIFO_KHR,"
+            "The application will use VK_PRESENT_MODE_FIFO_KHR present mode in vkCreateSwapchainKHR, which may"
```
Comment (fabian-lunarg) [inline framework/encode/vulkan_capture_manager.cpp:2718]: just a small remark. it looks like it's missing a space at the end of the line: "... which maycause ..."

**Comment by bradgrantham-lunarg** [comment]
This sounds like a good PR.  Because it changes the default capture behavior and some developers may have depended on the behavior (maybe without knowing it), we would like a little time to let this soak and discuss it with a couple of high-priority developers.  We anticipate this going in late next week if no one has any objections.

File: USAGE_android.md (hunk 1 above)
Comment (zongdu-arm) [reply USAGE_android.md]: I have fixed it, thanks.

File: framework/encode/vulkan_capture_manager.cpp (hunk 2 above)
Comment (zongdu-arm) [reply framework/encode/vulkan_capture_manager.cpp]: I have fixed it, thanks.

**Comment by MarkY-LunarG** [comment]
> This sounds like a good PR. Because it changes the default capture behavior and some developers may have depended on the behavior (maybe without knowing it), we would like a little time to let this soak and discuss it with a couple of high-priority developers. We anticipate this going in late next week if no one has any objections.

Clarification, Brad did a quick once over and had not realized that the behavior is already there and done by default, but this option is actually allowing the default behavior to be changed for anyone who wants to modify it with this new setting.  Because of that, I believe we're good to go.

**Review by bradgrantham-lunarg — APPROVED** [review]

**Comment by bradgrantham-lunarg** [comment]
Just a moment - I realize that since this has changed recently I need to review again

**Review by bradgrantham-lunarg — APPROVED** [review]

**Pull request merged**